modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c

```diff
@@ -44,8 +44,29 @@ static void hfclk_on_callback(struct onoff_manager *mgr,
 }
 
 #if defined(CONFIG_CLOCK_CONTROL_NRF)
+
+#if defined(NRF54LM20A_ENGA_XXAA)
+/* HF clock time to ramp-up. */
+#define MAX_HFXO_RAMP_UP_TIME_US 550
+
+static void hfclk_started_timer_handler(struct k_timer *dummy)
+{
+	hfclk_on_callback(NULL, NULL, 0, 0);
+}
+
+K_TIMER_DEFINE(hfclk_started_timer, hfclk_started_timer_handler, NULL);
+#endif
+
 void nrf_802154_clock_hfclk_start(void)
 {
+#if defined(NRF54LM20A_ENGA_XXAA)
+	/*
+	 * Right now, the power_clock_irq is not available on nRF54LM20A.
+	 * Since the onoff mechanism relies on the irq, the timer is used
+	 * to emit the hfclk_ready callback.
+	 */
+	k_timer_start(&hfclk_started_timer, K_USEC(MAX_HFXO_RAMP_UP_TIME_US), K_NO_WAIT);
+#endif
```

Comment:
`nrf_802154_clock_hfclk_start` might be called also from other priorities than the highest.
Because of that I would recommend to start the `hfclk_started_timer` after the `int ret = onoff_request(mgr, &hfclk_cli);` in this function. Otherwise the timer is started and counting, but the clock is not started yet. Any irq of high enough priority can lead to a situation when the `hfclk_started_timer_handler` fires too early because the call to `onoff_request` got delayed.